applications/lookup_table/lookup_table.py

```diff
@@ -9,17 +11,26 @@ def slowfun_too_slow(x, y):
 
     return v
 
+
+cache = {}
+
+
 def slowfun(x, y):
     """
     Rewrite slowfun_too_slow() in here so that the program produces the same
     output, but completes quickly instead of taking ages to run.
     """
     # Your code here
 
+    tuple = (x, y)
+    if tuple in cache:
+        return cache[tuple]
```

Comment:
Having the ability to save a whole pair as a key in and of itself can prove handy.